src/Header.js

```diff
@@ -53,8 +61,7 @@ class Header extends Component {
             </button>
         </div>
         <ul className="headerList">
-            <li><Link to='/profile'>Profile</Link></li>
-            <li><Link to='/login'>Login</Link></li>
+            <LoginOrProfileOption />
```

Comment:
I think this has to be { LoginOrProfileOption(this.state.isUserLoggedIn) } which will resolve into the JSX.   It could could also be written inline instead of with a method.